test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -368,14 +423,159 @@ def test_generator(setup_and_teardown):
+                     "assembly_path": job_dir / ASSEMBLY_NAMES[0],
+                     "assembly_info": assembly_obj_infos[0],
+                     "genome_info": genome_obj_infos[0]},
+                    {"genome_ref": "42/4/1",
+                     "assembly_ref": "42/2/1",
+                     "assembly_path": job_dir / ASSEMBLY_NAMES[1],
+                     "assembly_info": assembly_obj_infos[1],
+                     "genome_info": genome_obj_infos[1]}]
+    }
+    gfu.genbanks_to_genomes.return_value = genbanks_to_genomes_results
+
+    with patch.object(workspace_uploader, '_JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER', new=job_dir):
+        uploaded_count = workspace_uploader._upload_objects_in_parallel(
+            ws=ws,
+            upload_env_key="CI",
+            workspace_id=42,
+            load_id="214",
+            collections_source_dir=collection_source_dir,
+            wait_to_upload_objs=wait_to_upload_genomes,
+            batch_size=2,
+            source_data_dir=sourcedata_dir,
+            asu_client=Mock(),
+            gfu_client=gfu,
+            job_data_dir=job_dir,
+            upload_assembly_only=False,
+        )
+
+    assert uploaded_count == 2
+
+    # assert that no interactions occurred with ws
+    ws.get_object_info3.assert_not_called()
+
+    # assert that asu was called correctly
+    gfu.genbanks_to_genomes.assert_called_once_with(
+        {
+            "workspace_id": 42,
+            "inputs": [
+                {
+                    "file": {"path": f"{job_dir}/GCF_000979855.1_gtlEnvA5udCFS_genomic.gbff.gz"},
+                    "genome_name": "GCF_000979855.1_gtlEnvA5udCFS_genomic.gbff.gz",
+                    "metadata": {"load_id": "214"},
+                },
+                {
+                    "file": {"path": f"{job_dir}/GCF_000979175.1_gtlEnvA5udCFS_genomic.gbff.gz"},
+                    "genome_name": "GCF_000979175.1_gtlEnvA5udCFS_genomic.gbff.gz",
+                    "metadata": {"load_id": "214"},
+                }
+            ]
+        }
+    )
+
+    # check softlink for post_process
+    assert os.readlink(os.path.join(collection_source_dir, "42_1_1")) == os.path.join(
+        sourcedata_dir, "42_1_1"
+    )
+    assert os.readlink(os.path.join(collection_source_dir, "42_2_1")) == os.path.join(
+        sourcedata_dir, "42_2_1"
+    )
+
+    # check hardlink for post_process
+    assert os.path.samefile(
+        genome_collection_source_dirs[0] / ASSEMBLY_NAMES[0],
+        os.path.join(sourcedata_dir, "42_1_1", "42_1_1.fna.gz")
+    )
+
+    assert os.path.samefile(
+        genome_collection_source_dirs[1] / ASSEMBLY_NAMES[1],
+        os.path.join(sourcedata_dir, "42_2_1", "42_2_1.fna.gz")
+    )
+
+    # check metadata file
+    metadata_file = sourcedata_dir / "42_1_1" / "42_1_1.meta"
+    with open(metadata_file, 'r') as file:
+        data = json.load(file)
+
+    expected_metadata = {
+        'upa': '42/1/1',
+        'name': 'assembly_1',
```

Comment:
assembly_1 or ASSEMBLY_NAMES[0]?

Reply:
this metadata is retrieved from obj_info. So this is 'assembly_1'. I updated the test, hopefully that makes it more clear. And once I am at introducing `file_name` to `WSObjTuple`, this will probably be more clear.

Reply:
Ok, it's just that if the values are different for the same thing it makes the test much more confusing

Reply:
👍 should be consistent now.

Reply:
:+1: